### PR TITLE
Add static HTTP file server for /.well-known/ endpoint

### PR DESCRIPTION
**Pull request by alex19srv**
First commit to implement /.well-known/security.txt endpoint (as discussed in #21942)

### CODE REVIEW

**Comment by denyskon** [comment]
Could we have a test for it?

**Comment by alex19srv** [comment]
> Could we have a test for it?

I want to add tests, when I will add /.well-known/security.txt. Currently I can only test, if https://gitea/.well-known/ will return 404 status code (do not make directory list).

Or which tests do You want to see? I thought and did not found which tests to implement an this stage.

File: routers/web/web.go [hunk 1]
```diff
@@ -349,6 +349,12 @@ func registerRoutes(m *web.Route) {
 		m.Get("/change-password", func(ctx *context.Context) {
 			ctx.Redirect(setting.AppSubURL + "/user/settings/account")
 		})
+		wellKnownDir := wellKnownWebDir()
+		m.Get("/*", func(ctx *context.Context) {
```
Comment (lunny) [inline routers/web/web.go]: I don't think this is right. Why not use `m.Get("/.well-know/*", `

Comment (alex19srv) [reply routers/web/web.go]: This code inside group `m.Group("/.well-known", func() {` and my code works (I checked). All endpoints related to `"/.well-known"` is inside that group.

Comment (alex19srv) [reply routers/web/web.go]: Moved `m.Get("/.well-known/*", func(ctx *context.Context) {` out of `Group`.

**Comment by wxiaoguang** [comment]
Some thoughts:

1. `makeDirExist` seems strange, I do not think the app needs to create the directories automatically.
    * Or, if it is necessary, should Gitea provide a default "security.txt"?
2. After reading more code, I find there are some other files like "robots.txt" which need to be customized.
    * At the moment, the "robots.txt" is hard-coded as "CustomPath/robots.txt", it's impossible to add more files in the root directory.
    * Ideally, Gitea could support various static files in the web root or web well-known directory.
    * Maybe a more general approach could be like this:
        *  Serve files in `CustomPath/web-root-static/...` to `http://gitea-domain/...`.
        * For files in the root
            * `CustomPath/web-root-static/robots.txt`
            * `CustomPath/web-root-static/the-domain-verification.txt`
        * For files in the well-known
            * `CustomPath/web-root-static/.well-known/security.txt`
            * `CustomPath/web-root-static/.well-known/the-domain-verification.txt`

**Comment by wxiaoguang** [comment]
This feature is useful IMO, eg: https://discourse.gitea.io/t/well-known-path-serving-custom-files-behind-proxy/5445

I will try to propose a new PR for it.

**Comment by wxiaoguang** [comment]
->  Add "security.txt" support, add CORS header for ".well-known" #25974

**Comment by n0toose** [comment]
This is meta, but you should _seriously_ start providing some sort of method for people that are intending to contribute seriously and want to discuss design ideas and plans in a way that isn't a real-time platform.

**Comment by jolheiser** [comment]
> This is meta, but you should _seriously_ start providing some sort of method for people that are intending to contribute seriously and want to discuss design ideas and plans in a way that isn't a real-time platform. There are indeed too many issues but some sort of an explicitly defined "lifeline" to a contributor has to somehow continue existing.

I'm not sure what this means? Anyone is welcome to discuss ideas in an issue, Discord, Matrix, Discourse, etc.

**Comment by wxiaoguang** [comment]
Some people just don't want to discuss. There is a discord channel `develop`.

And some tasks are too difficult for new contributors. I have done my best to help them (many discussions were done by discord develop channel or private message)

**Comment by n0toose** [comment]
> Some people just don't want to discuss.

Issues are big and messy and there is an implicit expectation from the perspective of a new contributor that doesn't want to bother: Developers can't be possibly monitoring a thousand issues (which are arguably a very good place for drive-by contributors), even if notifications exist.

> And some tasks are too difficult for new contributors. I have done my best to help them (many discussions were done by discord develop channel or private message)

I'm not accusing you of not trying to help and not trying to help onboard new users (I apologize if I gave off that impression), I'm saying that it would be best to bridge the gap between wanting to work on an issue and reaching out to people. If you are reachable in an issue tracker, it might be best to not let that be an unwritten rule.

**Pull request closed without merging**